test/test_utils/__init__.py

```diff
@@ -650,11 +650,30 @@ def get_framework_and_version_from_tag(image_uri):
             f"Cannot find framework in image uri {image_uri} " f"from allowed frameworks {allowed_frameworks}"
         )
 
-    tag_framework_version = image_uri.split(":")[-1].split("-")[0]
+    tag_framework_version = re.search(":(.*?)-", image_uri).groups()[0]
```

Comment:
Thanks for this change. Could you make the framework version expression more specific by changing it to something like `\d+(\.\d+){2}`?

Also, please use raw strings for regular expressions, by placing an `r` before the string literal. For example, `r"(\d+(\.\d+){2})"`

Reply:
Addressed